x/ibc-transfer/handler_test.go

```diff
@@ -3,309 +3,64 @@ package transfer_test
-
-	trustingPeriod time.Duration = time.Hour * 24 * 7 * 2
-	ubdPeriod      time.Duration = time.Hour * 24 * 7 * 3
-	maxClockDrift  time.Duration = time.Second * 10
-)
-
-// define variables used for testing
-var (
-	testAddr1 = sdk.AccAddress([]byte("testaddr1"))
-	testAddr2 = sdk.AccAddress([]byte("testaddr2"))
-
-	testCoins, _          = sdk.ParseCoins("100atom")
-	testPrefixedCoins1, _ = sdk.ParseCoins(fmt.Sprintf("100%satom", types.GetDenomPrefix(testPort1, testChannel1)))
-	testPrefixedCoins2, _ = sdk.ParseCoins(fmt.Sprintf("100%satom", types.GetDenomPrefix(testPort2, testChannel2)))
+	ibctesting "github.com/cosmos/cosmos-sdk/x/ibc/testing"
 )
 
 type HandlerTestSuite struct {
 	suite.Suite
 
-	cdc *codec.Codec
+	coordinator *ibctesting.Coordinator
 
-	chainA *TestChain
-	chainB *TestChain
+	// testing chains used for convenience and readability
+	chainA *ibctesting.TestChain
+	chainB *ibctesting.TestChain
 }
 
 func (suite *HandlerTestSuite) SetupTest() {
-	suite.chainA = NewTestChain(testClientIDA)
-	suite.chainB = NewTestChain(testClientIDB)
-
-	suite.cdc = suite.chainA.App.Codec()
+	suite.coordinator = ibctesting.NewCoordinator(suite.T(), 2)
+	suite.chainA = suite.coordinator.GetChain(ibctesting.GetChainID(0))
+	suite.chainB = suite.coordinator.GetChain(ibctesting.GetChainID(1))
 }
 
+// constructs a send from chainA to chainB on the established channel/connection
+// and sends the coins back from chainB to chainA.
 func (suite *HandlerTestSuite) TestHandleMsgTransfer() {
-	handler := transfer.NewHandler(suite.chainA.App.TransferKeeper)
-
-	// create channel capability from ibc scoped keeper and claim with transfer scoped keeper
-	capName := host.ChannelCapabilityPath(testPort1, testChannel1)
-	cap, err := suite.chainA.App.ScopedIBCKeeper.NewCapability(suite.chainA.GetContext(), capName)
-	suite.Require().Nil(err, "could not create capability")
-	err = suite.chainA.App.ScopedTransferKeeper.ClaimCapability(suite.chainA.GetContext(), cap, capName)
-	suite.Require().Nil(err, "transfer module could not claim capability")
-
-	ctx := suite.chainA.GetContext()
-	msg := types.NewMsgTransfer(testPort1, testChannel1, testPrefixedCoins2, testAddr1, testAddr2.String(), 110, 0)
-	res, err := handler(ctx, msg)
-	suite.Require().Error(err)
-	suite.Require().Nil(res, "%+v", res) // channel does not exist
+	clientA, clientB, _, _, channelA, channelB := suite.coordinator.Setup(suite.chainA, suite.chainB)
+	handlerA := ibctransfer.NewHandler(suite.chainA.App.TransferKeeper)
 
-	// Setup channel from A to B
-	suite.chainA.CreateClient(suite.chainB)
-	suite.chainA.createConnection(testConnection, testConnection, testClientIDB, testClientIDA, connectiontypes.OPEN)
-	suite.chainA.createChannel(testPort1, testChannel1, testPort2, testChannel2, channeltypes.OPEN, channeltypes.ORDERED, testConnection)
+	coinToSendToB := sdk.NewCoins(sdk.NewCoin(fmt.Sprintf("%s/%s/%s", channelB.PortID, channelB.ID, sdk.DefaultBondDenom), sdk.NewInt(100)))
```

Comment:
this could be using a helper func, but will probably get rewritten with the new ibc-transfer denom format